src/Storage/Repository/PgInstanceRepository.cs

```diff
@@ -68,137 +86,176 @@ public async Task<bool> Delete(Instance item)
             string continuationToken,
             int size)
         {
-            InstanceQueryResponse queryResponse = new() { Count = 0, Instances = new() };
-            long continueIdx = string.IsNullOrEmpty(continuationToken) ? -1 : long.Parse(continuationToken);
-            int maxIterations = 100_000;
-            int currentIteration = 0;
-            while (queryResponse.Count < size)
+            try
             {
-                if (++currentIteration > maxIterations)
-                {
-                    queryResponse.Exception = "Please narrow the seach parameters. The current search is too slow.";
-                    return queryResponse;
-                }
+                return await GetInstancesInternal(queryParams, continuationToken, size);
+            }
+            catch (Exception e)
+            {
+                return new() { Count = 0, Instances = new(), Exception = e.Message };
```

Comment:
We can improve this from how it was by logging the exception. As it stands we only keep the exception message.

Reply:
Changed